samples/markdown-transforms/format-output-blocks.sed

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/sed -Enf
+
+# nbconvert formats output blocks as idented sections of Markdown.
+# We recognise these (with 4 spaces per indentation level),
+# remove this leading whitespace, then attempt to infer the type of
+# output as JSON, YAML, or simply a generic block.
+
+################################################################################
+
+# Ignore indentation within code blocks.
+
+# If we find a starting delimiter, record that we're in a code block.
+/^```.+/ {
+    h
+    p
+    b END
+}
+
+# If we find an ending delimiter, stop acting like we're in a code block.
+/^```$/ {
+    p
+    z
+    h
+    b END
+}
+
+# Check if we're in a code block.
+# If so, print this line and skip to the next.
+x
+/^```/ {
+    x
+    p
+    b END
+}
+
+# Otherwise, keep investigating the current line.
+x
+
+################################################################################
+
+# If we're within an output block...
+/^\s{4}.*/ {
+  s/^\s{4}//
+  H
+  b END
+}
+
+# We're outside an output block
+x
+
+# Special case for unindented links
+/^.+$/ {
```

Comment:
:wrench: Ideally we would render this link in the normal font rather than as monospace